pennylane/ops/qubit/state_preparation.py

```diff
@@ -185,23 +185,32 @@ def state_vector(self, wire_order: Optional[WiresLike] = None) -> TensorLike:
         return math.convert_like(ket, prep_vals)
 
 
-def _basis_state_decomp_resources(state, wires):
-    if not qml.math.is_abstract(state):
-        return {qml.X: len([wire for wire, basis in zip(wires, state, strict=True) if basis == 1])}
-    return {qml.PhaseShift: 2 * len(wires), qml.RX: len(wires)}
+def _basis_state_decomp_resources(num_wires):
+    # Represent one of the X gates as an RX and a GlobalPhase because RX is
+    # used when jax-jit is enabled without capture/qjit.
+    return {qml.X: num_wires, qml.RX: 1, qml.GlobalPhase: 1}
```

Comment:
Why only one here?

Reply:
This is a hack, we have to account for the decomposition potentially producing `RX` gates, but at the same time we don't want to affect the overall resource estimate associated with this operator.